torch/library.py

```diff
@@ -149,6 +197,64 @@ def wrap(f):
     return wrap
 
 
+def impl_fake(lib, op_name, *, _stacklevel=3):
```

Comment:
API design help wanted please: I'm not sure what the torch.library API for registering a FakeTensor implementation should be. Right now, the API is analogous to the existing torch.library.Library.impl and torch.library.impl, APIs in that they accept a torch.Library object.

However, we don't actually *need* the torch.library object, because the FakeTensor implementation isn't tied to a dispatch key! An alternate API (inspired by register_meta) could be look like the following:
```py
def impl_fake(op: OpOverload, ...): ...

@impl_fake(torch.ops.custom.foo)
def foo_fake(ctx, x):
    ...
```

If we prefer something more like this I can change it. The other considerations are that we will end up adding an `impl_autograd`, and `impl_autograd` may need to be tied to a torch.Library object.

Reply:
So remember we were talking about the mid level vs high level API?

IMO, it doesn't make any sense to have fake tensor registrations in the mid level API, *unless* you want to add a fake tensor dispatch key (which is plausible, but we'd have to work out the consequences.) Mid level API is "stuff on top of existing torch.Library", tehrefore fake tensor shouldn't work here.